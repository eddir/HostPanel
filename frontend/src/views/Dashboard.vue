<template>
  <div>
    <CRow>
      <CCol md="12">
        <servers></servers>
      </CCol>
    </CRow>
  </div>
</template>

<script>
import Servers from "@/views/servers/Servers"

export default {
  name: 'Dashboard',
  components: {Servers},
}
</script>
