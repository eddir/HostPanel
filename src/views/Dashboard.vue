<template>
  <div>
    <CRow>
      <CCol md="12">
        <CCard>
          <CCardHeader>
            Сервера
          </CCardHeader>
          <CCardBody>
            <servers></servers>
          </CCardBody>
        </CCard>
        <CCard>
          <CCardBody>
            <CCardHeader>Вход через телеграм ({{telegramBot}})</CCardHeader>
            <CCardBody>
              <vue-telegram-login
                  mode="callback"
                  :telegram-login="telegramBot"
                  @callback="telegramAuth" />
            </CCardBody>
          </CCardBody>
        </CCard>
      </CCol>
    </CRow>
  </div>
</template>

<script>
import {vueTelegramLogin} from 'vue-telegram-login'

import Servers from "@/views/servers/Servers";
import api from "@/services/API"

export default {
  name: 'Dashboard',
  components: {Servers, vueTelegramLogin},
  data() {
    return {
      telegramBot: api.AUTH_TELEGRAM_BOT
    }
  },
  methods: {
    telegramAuth(user) {
      api.login(user);
    }
  }
}
</script>
