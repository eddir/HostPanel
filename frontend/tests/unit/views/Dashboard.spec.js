import Vue from 'vue'
import regeneratorRuntime from "regenerator-runtime";
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Dashboard from '@/views/Dashboard'


Vue.use(regeneratorRuntime)
Vue.use(CoreuiVue)

describe('Dashboard.vue', () => {
  it('has a name', () => {
    expect(Dashboard.name).toBe('Dashboard')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Dashboard)
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Dashboard', () => {
    const wrapper = shallowMount(Dashboard)
    expect(wrapper.findComponent(Dashboard)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Dashboard)
    expect(wrapper.element).toMatchSnapshot()
  })
})
