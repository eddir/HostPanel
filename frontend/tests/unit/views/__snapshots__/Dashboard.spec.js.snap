// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="12"
      tag="div"
    >
      <ccard-stub>
        <ccardheader-stub>
          
          Сервера
        
        </ccardheader-stub>
         
        <ccardbody-stub>
          <servers-stub />
        </ccardbody-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
</div>
`;
